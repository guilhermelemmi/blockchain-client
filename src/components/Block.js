import React from "react";
import PropTypes from "prop-types";
import { Typography, ListItem, makeStyles } from "@material-ui/core";
import colors from "../constants/colors";

const Block = ({ index, data }) => {
  const classes = useStyles();
  return (
    <ListItem className={classes.root} disableGutters>
      <Typography variant="h6" className={classes.heading}>
        {parseInt(index, 10).toLocaleString("en", { minimumIntegerDigits: 3 })}
      </Typography>
      <Typography className={classes.content}>{data}</Typography>
    </ListItem>
  );
};

const useStyles = makeStyles((theme) => ({
  root: {
    backgroundColor: "rgba(0,0,0,.12)",
    borderRadius: 2,
    marginBottom: 4,
    padding: 8,
    alignItems: "flex-start",
    flexDirection: "column",
    "&:last-child": {
      marginBottom: 0,
    },
  },
  content: {
    fontSize: theme.typography.pxToRem(14),
    color: colors.text,
    lineHeight: 1,
  },
  heading: {
    fontSize: theme.typography.pxToRem(10),
    fontWeight: "bold",
    color: colors.blue,
    lineHeight: "16px",
    marginBottom: 4,
    textTransform: "uppercase",
    letterSpacing: 2,
  },
}));

Block.propTypes = {
  index: PropTypes.string.isRequired,
  data: PropTypes.string.isRequired,
};

export default Block;
