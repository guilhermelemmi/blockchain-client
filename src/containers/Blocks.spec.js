import React from "react";
import { shallow } from "enzyme";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { Provider } from "react-redux";
import { create } from "react-test-renderer";
import { CircularProgress } from "@material-ui/core";
import ConnectedBlocks, { Blocks } from "./Blocks";
import Block from "../components/Block";

describe("<Blocks />", () => {
  const actions = {
    fetchNodeBlocks: jest.fn(),
  };

  const nodeA = {
    url: "https://thawing-springs-53971.herokuapp.com",
    online: false,
    name: "Node 1",
    blocks: ["hashA", "hashB"],
  };

  const blocks = {
    byId: {
      hashA: {
        hash: "hashA",
        prevHash: "",
        data: "dataA",
        index: "1",
      },
      hashB: {
        hash: "hashB",
        prevHash: "hashA",
        data: "dataB",
        index: "2",
      },
    },
    allIds: ["hashA", "hashB"],
  };

  it("should contain <Block />", () => {
    const wrapper = shallow(
      <Blocks
        actions={actions}
        node={nodeA}
        blocks={blocks.allIds.map((blockId) => blocks.byId[blockId])}
        error={false}
        loading={false}
      />
    );

    expect(wrapper.find(Block).length).toEqual(2);
  });

  it("should show loading progress", () => {
    const wrapper = shallow(
      <Blocks
        actions={actions}
        node={nodeA}
        blocks={[]}
        error={false}
        loading={true}
      />
    );

    expect(wrapper.find(CircularProgress).length).toEqual(1);
  });

  it("should show error message", () => {
    const wrapper = shallow(
      <Blocks
        actions={actions}
        node={nodeA}
        blocks={[]}
        error={true}
        loading={false}
      />
    );
    const div = wrapper.find("div");
    expect(div.text()).toEqual("Oops, something wrong happened! :(");
  });

  it("should match snapshot", () => {
    const middlewares = [thunk];
    const store = configureMockStore(middlewares)({
      actionStatuses: { fetchNodeBlocks: {} },
      entities: { blocks },
    });
    const component = create(
      <Provider store={store}>
        <ConnectedBlocks actions={actions} node={nodeA} />
      </Provider>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
