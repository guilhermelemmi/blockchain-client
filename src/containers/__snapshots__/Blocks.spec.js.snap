// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Blocks /> should match snapshot 1`] = `
<ul
  className="MuiList-root MuiList-padding"
  divider={true}
  style={
    Object {
      "flex": 1,
      "padding": 0,
    }
  }
>
  <li
    className="MuiListItem-root makeStyles-root-1"
    disabled={false}
  >
    <h6
      className="MuiTypography-root makeStyles-heading-3 MuiTypography-h6"
    >
      001
    </h6>
    <p
      className="MuiTypography-root makeStyles-content-2 MuiTypography-body1"
    >
      dataA
    </p>
  </li>
  <li
    className="MuiListItem-root makeStyles-root-1"
    disabled={false}
  >
    <h6
      className="MuiTypography-root makeStyles-heading-3 MuiTypography-h6"
    >
      002
    </h6>
    <p
      className="MuiTypography-root makeStyles-content-2 MuiTypography-body1"
    >
      dataB
    </p>
  </li>
</ul>
`;
